### PR TITLE
Extract step state tracking into its own module.

### PR DESCRIPTION
**Pull request by kyrofa**
This PR makes progress on LP: [#1477904](https://bugs.launchpad.net/snapcraft/+bug/1477904), and will be utilized in subsequent commits that further flesh out the states.

### CODE REVIEW

File: snapcraft/pluginhandler.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
 #
-# Copyright (C) 2015 Canonical Ltd
+# Copyright (C) 2016 Canonical Ltd
```
Comment (sergiusens) [inline snapcraft/pluginhandler.py]: It's 2015, 2016 or 2015-2016

File: snapcraft/pluginhandler.py [hunk 2]
```diff
@@ -33,6 +33,7 @@
     common,
     libraries,
     repo,
+    states,
```
Comment (sergiusens) [inline snapcraft/pluginhandler.py]: Call this `_states`. Let's start making sure private suff it private.

File: snapcraft/pluginhandler.py (hunk 1 above)
Comment (kyrofa) [inline snapcraft/pluginhandler.py]: Argh, I knew that.

File: snapcraft/pluginhandler.py (hunk 2 above)
Comment (kyrofa) [inline snapcraft/pluginhandler.py]: Good idea.

**Comment by sergiusens** [comment]
Looks good, just fix the copyright dates :smile:

File: snapcraft/pluginhandler.py (hunk 1 above)
Comment (kyrofa) [inline snapcraft/pluginhandler.py]: Done.

**Comment by sergiusens** [comment]
rebase/update

**Comment by kyrofa** [comment]
> rebase/update

Done.

File: snapcraft/internal/states/__init__.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from snapcraft.internal.states.strip_state import StripState  # noqa
+from snapcraft.internal.states.stage_state import StageState  # noqa
```
Comment (sergiusens) [inline snapcraft/internal/states/__init__.py]: I thought we agreed on 
```python
from snapcraft.internal.states._strip_state import StripState
from snapcraft.internal.states._stage_state import StageState
```
?

Comment (kyrofa) [inline snapcraft/internal/states/__init__.py]: Ah right, done-- sorry!

**Pull request merged**